components/channel_layout/channel_identifier_router/actions.ts

```diff
@@ -174,7 +174,7 @@ export function goToChannelByChannelName(match: Match, history: History): Action
             const user = getCurrentUser(getState());
             const isSystemAdmin = Utils.isSystemAdmin(user?.roles);
             if (isSystemAdmin) {
-                if (channel.type === Constants.PRIVATE_CHANNEL) {
+                if (channel && channel.type === Constants.PRIVATE_CHANNEL) {
```

Comment:
Thank you @streamer45  .. A Small suggestion 🙂 
```suggestion
                if (channel?.type === Constants.PRIVATE_CHANNEL) {
```

Reply:
Nice. Is this typescript specific or newly introduced in JS as well?

Reply:
Yeah it makes code look allot cleaner. It's a [JS](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) one.

Reply:
Definitely. Thanks for that :+1: